app/services/fix_provider_audits.rb

```diff
@@ -0,0 +1,25 @@
+class FixProviderAudits
+  def initialize
+    @provider_users = {}
+  end
+
+  def call
+    Audited::Audit.where("username like '%(Provider)'").find_each do |audit|
+      email_address = email_address_from(audit)
+      provider_user = find_provider_user_by(email_address)
+      audit.update(user: provider_user) if provider_user
```

Comment:
What should we do if there's no user found? It would be good if we could depend on there being something there

Reply:
Going to check prod to see whether there are any in this state

Reply:
Have checked prod in the following way

```ruby
found_users = Audited::Audit.where('username LIKE ?', '%Provider%')
  .map { |a| a.username.split(' (Provider)').first }
  .uniq
  .map { |e| ProviderUser.find_by(email_address: e) }
```

There are no `nil`s in this array so all the emails have corresponding `ProviderUsers`.